source/configuration_manual/mail_cache_settings.rst

```diff
@@ -31,4 +37,93 @@ The list of cached fields depends on which IMAP clients are expected to be used.
+   another system where the IMAP and POP3 messages' order differs.
+ * ``guid``: Internal Dovecot GUID for messages.
+ * ``mime.parts``: MIME parts' sizes.
+ * ``binary.parts``: MIME parts' sizes after decoding Content-Transfer-Encoding
+   to binary. Used by IMAP BINARY extension.
+ * ``body.snippet``: A short snippet (IMAP PREVIEW) of the message body.
+ * ``hdr.*``: Message headers listed individually, e.g. ``hdr.date``,
+   ``hdr.from``
+
+Cache decision algorithm
+------------------------
+
+Nowadays there are three types of IMAP clients:
+
+ * Clients that download all the (new) messages' headers and other metadata
+   into local cache. Afterwards they use only the local cache for accessing
+   the metadata. They may download the message bodies later on though. These
+   clients don't benefit from Dovecot's caching after they have downloaded
+   the metadata. For example Outlook, Thunderbird, OSX Mail.
+ * Clients that work otherwise like the local clients described above, but
+   don't download all mails locally (even their metadata). They may also use
+   server-side search for mails. For example iOS Mail.
+ * Clients that don't have a local cache at all, or only a short-lived cache.
+   These clients would benefit from having all mails cached by Dovecot.
+   For example webmails.
+
+IMAP clients don't advertise how they work, so Dovecot attempts to figure it
+out dynamically. The behavior is now:
+
+ * For a newly created INBOX Dovecot gets the caching decisions from the
+   :ref:`setting-mail_cache_fields` and :ref:`setting-mail_always_cache_fields`
+   settings.
+ * For a newly created non-INBOX folder the caching decisions are copied from
+   the INBOX.
+ * Whenever a new non-cached field is accessed, its caching decision is set to
+   TEMP, which means only the last 1 week's mails have the field cached.
+ * Whenever a field is accessed for a mail older than 1 week the caching
+   decision changes from TEMP to YES, which means the field is cached for all
+   mails.
+ * Whenever a single IMAP session accesses the mails in non-ascending order
+   (e.g. mail UIDs 100 -> 99 instead of 99 -> 100) the caching decision changes
+   from TEMP to YES. Especially the SORT/THREAD commands trigger this.
+ * Whenever cache file is recreated (purged) it can cause some fields'
+   decisions to change:
+
+    * .. versionchanged:: v2.3.11 Changes YES -> TEMP if the YES decision hasn't
+                          been reconfirmed for the last 30 days
+                          (:ref:`setting-mail_cache_unaccessed_field_drop`).
+			  Older versions changed the YES -> TEMP decision every
+			  time the cache was purged, which could have happened
+			  too early sometimes.
+    * .. versionchanged:: v2.3.11 Changes TEMP -> NO and drops the field if it
+                          hasn't been accessed for the last 60 days (2 *
+                          :ref:`setting-mail_cache_unaccessed_field_drop`).
+			  Older versions dropped it after 30 days (1 *
```

Comment:
seems like indent is different here.

Reply:
Well, it's github showing tabs badly. I think Sphinx complains if the indent isn't exactly right. Although I suppose it would be better if there weren't tabs in the first place. Changed.